docs/cli/dotnet-test.md

```diff
@@ -0,0 +1,137 @@
+dotnet-test
+================
+
+## NAME
+
+`dotnet-test` - runs unit test using the configured test runner
+
+## SYNOPSIS
+
+dotnet-test [--configuration]  
+    [--output] [--build-base-path] [--framework] [--runtime]
+    [--no-build]
+    [--parentProcessId] [--port]  
+    [< project >]  
+
+## DESCRIPTION
+
+`dotnet-test` is used to execute unit tests in a given project. Unit tests are class library 
+projects that have dependencies on the unit test framework (e.g. NUnit or xUnit) and the 
+dotnet test runner for that unit testing framework. These are packaged as NuGet packages and are 
+restored as ordinary dependencies for the project.
+
+Test projects also need to specify a test runner property in project.json using the "testRunner" node. 
+This value should contain the name of the unit test framework.
+
+Below is a sample project.json that shows the needed properties:
+
+```json
+{
+    "version": "1.0.0-*",
+
+    "dependencies": {
+        "NETStandard.Library": "1.0.0-*",
+        "xunit": "2.1.0-*",
+        "dotnet-test-xunit": "1.0.0-*"
+    },
+    "testRunner": "xunit",
+
+    "frameworks": {
+        "netstandard1.5": {
+        }
+    }
+}
+```
+Dotnet test supports two running modes:
+
+1. Console: In console mode, dotnet test simply executes fully whatever command gets passed to it and outputs the results. Anytime you
+invoke dotnet test without passing --port, it will run in console mode, which in turn will cause the runner to run in console mode.
+2. Design time: Anytime you pass a port to dotnet test, we will run in design time. That means that dotnet test will connect to that port
+using TCP and will then exchange a established set of messages with whatever else is connected to that port. When this happens, the runner
+also receives a port (a new one, mind you) that dotnet test will use to communicate with it. The reason why the runner also uses TCP to
+communicate with dotnet test is because in design mode, it is not sufficient to just output results to the console. We need to send the
+adapter structure messages containing the results of the test execution.
+
+### Communication protocol in design time.
+
```

Comment:
Frankly, this communication protocol stuff should be in its own document. It's only useful for implementers of harnesses. Someone trying to use 'dotnet test' on their code doesn't care what strings get sent back and forth to a UI they probably aren't using anyway, because they typed 'dotnet test'.